### ISSUE TITLE
Pod deletion monitoring not working as intended?

### ISSUE THREAD

**Issue by litzk** [txn2/kubefwd#169]
Hey guys! First of all, love the work on kubefwd, it's a great tool!

Currently trying to make this part of the documentation work:
> Supports Pod-level forwarding monitoring. When the forwarded Pod is deleted (such as updating the deployment, etc.), the forwarding of the service to which the pod belongs is automatically restarted;

I tried with a "normal" service with ClusterIP and a headless service. Both didn't work. Didn't try forwarding to a pod directly but to my understanding that won't work.

What I could find:

On release [1.11.0](https://github.com/txn2/kubefwd/releases/tag/1.11.0) there's a mention to [this commit](https://github.com/txn2/kubefwd/commit/1ea20e25be7f40e9dae363e84c17a71d3dfae421). My Go knowledge is very limited but I understand that's what implements the "listener".

Reading fwdport.go in the latest release 1.17.X I find that the call to ListenUntilPodDeleted is commented (https://github.com/txn2/kubefwd/blob/1.17.x/pkg/fwdport/fwdport.go#L144-L146).

Is there something I'm not seeing or is there another way to achieve the same behaviour? Just looking to "maintain" the connection after pod restarts.

Sorry for my broken english, not my first language.

**Comment by dobesv** [comment]
I have a similar issue, I have to restart kubefwd a lot otherwise it will sort of "lose connection" with pods that have been restarted or had a connection error and never try to connect again.

I am running directly from git `HEAD`

**Comment by dantheman999301** [comment]
This did used to partially work for me but it broke after this PR - https://github.com/txn2/kubefwd/pull/171

**Comment by omh1280** [comment]
Bump @cjimti is this intended behavior?

**Comment by cjimti** [comment]
> Bump @cjimti is this intended behavior?

TL;DR: Yes, for the most part. 

I built `kubefwd` for myself and my team to build new services locally that connect to existing services in a cluster. These existing services should be relatively stable since they are the dependencies of the new services we are developing. Often we are tailing the logs of the pods kubefwd connects us to. Some services could have many pods backing them and it is helpful to know which one we are connected to for monitorign purposes.

A couple of issues with re-connect: 1) It's hard to determine what constitutes a failure other than the Pod itself going away, a Pod could simply stop listening on a port or reject connection for a brief period of time and this could be intended. 2) If the Pod crashed (or OOM, drained, etc..) then it may have been a result of our development against it, and silently re-connecting to a new Pod would hide that issue.

We have tried this re-connect functionality in the past and found it very difficult to make stable: 1) Have to watch/monitor each pod for events, 2) wait for a replacement Pod on the backing service to reach a Ready state, 2) assume the Pod has decent readiness probes and is truly ready to accept port-forwards; 2) Properly communicate these state changes to the user. 3) Create another feature flag since not all users desire this.

I add and restart services on the cluster occasionally or even have services fail; all it requires is a `ctrl-c`, `up-arrow` and `enter`, and I am back in business. 

I think some people might be trying to push `kubefwd` beyond its core use case, and this is cool, but I have to be careful not to destabilize or overly complicate its operation for everyone else.

Also, I don't have the resources to build a robust test suite, and as the complexity grown with each feature PR the hand testing is becoming overwhelming. Many people submit PRs and only test the functionality they use, unknowingly breaking other features they do not use. 

Unless this project can find some developers highly stilled in writing tests, documentation, and help in PR review, I not likely be adding many new features and instead focus on stability. I was too quick to merge some PRs in the past that caused problems for parts I don't use often, but others do.
